pkg/webhook/webhook.go

```diff
@@ -125,39 +116,15 @@ func (m *podMutator) Handle(ctx context.Context, req admission.Request) (respons
 			return admission.Errored(http.StatusBadRequest, err)
 		}
 		// bypass cache and get from the API server as it's not found in cache
-		err = m.reader.Get(ctx, types.NamespacedName{Name: serviceAccountName, Namespace: pod.Namespace}, serviceAccount)
-		if err != nil {
-			logger.Error(err, "failed to get service account")
-			return admission.Errored(http.StatusBadRequest, err)
+		if err = m.reader.Get(ctx, types.NamespacedName{Name: serviceAccountName, Namespace: pod.Namespace}, serviceAccount); err != nil {
+			logger.Error(err, "failed to get service account, using defaults")
```

Comment:
Should we remove this logic from the above cache'd get as well?

Reply:
In case of cache'd get we only return error if it's not `apierrors.IsNotFound` https://github.com/Azure/azure-workload-identity/blob/0ace9ecb69f10438ad6abe91970a8859d6e2cb60/pkg/webhook/webhook.go#L122-L125

For not found error, we try doing a get from API server and with this change even if that doesn't exist we just proceed without failing.

Reply:
Right I was more saying "maybe just ignore all errors here too just like in the live look up case"

Reply:
that sounds good!

Reply:
updated to ignore errors and also added text about `AZURE_CLIENT_ID` being empty in the error log.

Reply:
removed these changes as discussed!